tests/backup/backup_test.go

```diff
@@ -1011,467 +1025,765 @@ var _ = Describe("{BackupCrashVolDriver}", func() {
-		DisableSSL:       aws.Bool(disableSSLBool),
-		S3ForcePathStyle: aws.Bool(true),
-	},
-	)
-	Expect(err).NotTo(HaveOccurred(),
-		fmt.Sprintf("Failed to get S3 session to create bucket. Error: [%v]", err))
+		Step("Deploy applications", func() {
+			contexts = make([]*scheduler.Context, 0)
+			bkpNamespaces = make([]string, 0)
+			for i := 0; i < Inst().ScaleFactor; i++ {
+				taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+				logrus.Infof("Task name %s\n", taskName)
+				appContexts := ScheduleApplications(taskName)
+				contexts = append(contexts, appContexts...)
+				for _, ctx := range appContexts {
+					// Override default App readiness time out of 5 mins with 10 mins
+					ctx.ReadinessTimeout = appReadinessTimeout
+					namespace := GetAppNamespace(ctx, taskName)
+					bkpNamespaces = append(bkpNamespaces, namespace)
+				}
+			}
 
-	S3Client := s3.New(sess)
+			// Skip volume validation until other volume providers are implemented.
+			for _, ctx := range contexts {
+				ctx.SkipVolumeValidation = true
+			}
 
-	_, err = S3Client.CreateBucket(&s3.CreateBucketInput{
-		Bucket: aws.String(bucketName),
-	})
-	Expect(err).NotTo(HaveOccurred(),
-		fmt.Sprintf("Failed to create bucket [%v]. Error: [%v]", bucketName, err))
+			ValidateApplications(contexts)
+			for _, ctx := range contexts {
+				for vol, params := range GetVolumeParameters(ctx) {
+					volumeParams[vol] = params
+				}
+			}
+		})
 
-	err = S3Client.WaitUntilBucketExists(&s3.HeadBucketInput{
-		Bucket: aws.String(bucketName),
-	})
-	Expect(err).NotTo(HaveOccurred(),
-		fmt.Sprintf("Failed to wait for bucket [%v] to get created. Error: [%v]", bucketName, err))
-}
+		// TODO(stgleb): Add multi-namespace backup when ready in px-backup
+		for _, namespace := range bkpNamespaces {
+			go func(namespace string) {
+				backupName := fmt.Sprintf("%s-%s", backupNamePrefix, namespace)
+				Step(fmt.Sprintf("Create backup full name %s:%s:%s",
+					sourceClusterName, namespace, backupName), func() {
+					err = CreateBackupGetErr(backupName,
+						sourceClusterName, backupLocationName, backupLocationUID,
+						[]string{namespace}, labelSelectors, orgID)
+					if err != nil {
+						bkpNamespaceErrors[namespace] = err
+					}
+				})
+			}(namespace)
+		}
 
-func CreateAzureBucket(bucketName string) {
-	// From the Azure portal, get your Storage account blob service URL endpoint.
-	_, _, _, _, accountName, accountKey := getAzureCredsFromEnv()
+		var wg sync.WaitGroup
+		for _, namespace := range bkpNamespaces {
+			backupName := fmt.Sprintf("%s-%s", backupNamePrefix, namespace)
+			error, ok := bkpNamespaceErrors[namespace]
+			if ok {
+				logrus.Warningf("Skipping waiting for backup %s because %s", backupName, error)
+			} else {
+				wg.Add(1)
+				go func(wg *sync.WaitGroup, namespace, backupName string) {
+					defer wg.Done()
+					Step(fmt.Sprintf("Wait for backup %s to complete", backupName), func() {
+
+						ctx, err := backup.GetPxCentralAdminCtx()
```

Comment:
Move the `GetPxCentralAdminCtx` call outside the loop. That way we won't have to make this call every time.